### PR TITLE
Add Widmogrod namespace prefix everywhere. closes #23.

### LINKED ISSUE #23: Namespace clash with functional-php
Hello there,

Thanks for the great work on this library, it's nice to have some monad implementation especially based on FantasyLand :)

I have one not so small issue however. I usually like to use https://github.com/lstrojny/functional-php but sadly when installing both packages via composer there is a nameclash on the `invoke` that you both define in the `Functional` namespace.

```
PHP Fatal error:  Cannot redeclare Functional\invoke() (previously declared in /vagrant/src/vendor/lstrojny/functional-php/src/Functional/Invoke.php:37) in /vagrant/src/vendor/widmogrod/php-functional/src/Functional/functions.php on line 94
```

I am not sure what the best solution is. As fas as I know, it is not possible to fix this via composer. I would like to propose a renaming of your namespace as this library is probably less used than "functional-php" thus leading to fewer BC breaks.

I can see two possible renames :
- rename just the `Functional` namespace to something else, like `FunctionalHelpers` for examples. But there might be issues for other namespaces like `Monad` later.
- put everything "under" a parent namespace like `Widmogrod` or something like that.

What do you think ?

I can provide a PR doing all that as soon as we agree on the best solution.

Cheers,

### PR DESCRIPTION
**Pull request by krtek4**
This takes care of issue #23

I hope I didn't forget something. The test suite is running fine.

### CODE REVIEW

File: src/Monad/IO/functions.php [hunk 1]
```diff
@@ -157,15 +157,15 @@ function getArgs()
     return M\IO::of(function () {
         if (!ini_get('register_argc_argv')) {
             throw userError(sprintf(
-                'argv is not available, because ini option "register_argc_argv" is disabled'
+                'argv is not available, because Widmogrod\ini option "register_argc_argv" is disabled'
```
Comment (widmogrod) [inline src/Monad/IO/functions.php]:
``` diff
-'argv is not available, because Widmogrod\ini option "register_argc_argv" is disabled
+'argv is not available, because ini option "register_argc_argv" is disabled
```

File: src/Monad/Identity.php [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+    FantasyLand\Monad ,
+    ValueOfInterface
 {
-    const of = 'Monad\Identity::of';
+    const of = 'Widmogrod\Monad\Identity::of';
 
-    use Common\PointedTrait;
-    use Common\ValueOfTrait;
+    use PointedTrait;
+    use ValueOfTrait;
```
Comment (widmogrod) [inline src/Monad/Identity.php]: Using `namespaces` help helps me in understanding wether interface belong to a good place or not (especially during development), and as a side effect, there is less imports. If I may ask you to revert this change, at least for now.

File: src/Monad/Maybe/Just.php [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-use FantasyLand;
-use Applicative;
+use Widmogrod\Common\PointedTrait;
+use Widmogrod\Monad;
+use Widmogrod\FantasyLand;
 
 class Just implements Maybe
 {
-    use Common\PointedTrait;
+    use PointedTrait;
```
Comment (widmogrod) [inline src/Monad/Maybe/Just.php]: Same here. Please use `namespace` prefix.

File: src/Monad/State.php [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+use Widmogrod\FantasyLand\Apply;
+use Widmogrod\FantasyLand\Monad;
 
 class State implements Monad
 {
-    const of = 'Monad\State::of';
+    const of = 'Widmogrod\Monad\State::of';
 
-    use Common\PointedTrait;
+    use PointedTrait;
```
Comment (widmogrod) [inline src/Monad/State.php]: Same here. Please use `namespace` prefix.

**Review by widmogrod — CHANGES_REQUESTED** [review]
Thank you for your work.
Few small fixes remains and I going to merge this.

**Comment by krtek4** [comment]
Hey, 

Thanks for the feedback, I think everything is now corrected as you requested :)

File: src/Monad/Either/Left.php [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+use Widmogrod\Common\PointedTrait;
+use Widmogrod\Common\ValueOfTrait;
+use Widmogrod\FantasyLand;
 
 class Left implements Either
 {
-    use Common\PointedTrait;
-    use Common\ValueOfTrait;
+    use PointedTrait;
+    use ValueOfTrait;
```
Comment (widmogrod) [inline src/Monad/Either/Left.php]: Please use `namespace` prefix.

File: src/Monad/Either/Right.php [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+use Widmogrod\Common\PointedTrait;
+use Widmogrod\Common\ValueOfTrait;
+use Widmogrod\FantasyLand;
 
 class Right implements Either
 {
-    use Common\PointedTrait;
-    use Common\ValueOfTrait;
+    use PointedTrait;
+    use ValueOfTrait;
```
Comment (widmogrod) [inline src/Monad/Either/Right.php]: Please use `namespace` prefix.

File: src/Monad/IO.php [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
 
 class IO implements
     Monad,
     Foldable
 {
-    const of = 'Monad\IO::of';
+    const of = 'Widmogrod\Monad\IO::of';
 
-    use Common\PointedTrait;
+    use PointedTrait;
```
Comment (widmogrod) [inline src/Monad/IO.php]: Please use `namespace` prefix.

File: src/Monad/IO/functions.php [hunk 8]
```diff
@@ -56,15 +56,15 @@ function getChar()
     return M\IO::of(function () {
         if (false === ($char = fgetc(STDIN))) {
             throw userError(
-                'Can\'t read from stdin, because its closed'
+                'Can\'t read from stdin, because Widmogrod\its closed'
```
Comment (widmogrod) [inline src/Monad/IO/functions.php]:
``` diff
-Can\'t read from stdin, because Widmogrod\its closed
+Can\'t read from stdin, because its closed
```

**Review by widmogrod — CHANGES_REQUESTED** [review]
This PR is so huge, that I missed few places.

**Comment by krtek4** [comment]
Sorry, I should have also checked... I searched the code and it seems to me it's fine now.

**Comment by widmogrod** [comment]
@krtek4 I found last place where the changes are not reflected. README.md. Could you also update it?

**Review by widmogrod — APPROVED** [review]

**Comment by krtek4** [comment]
Done, I hope I didn't forget anything else this time... I will be a bit more careful next time.

**Comment by widmogrod** [comment]
@krtek4 Thank you a lot! Great work!
